scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -276,7 +290,8 @@ class Router(formatOps: FormatOps) {
           Split(Space, 0)
             .notIf(
               style.newlines.alwaysBeforeCurlyBraceLambdaParams ||
-                isSelfAnnotation || lambdaPolicy == null
+                isSelfAnnotation || lambdaPolicy == null ||
+                (style.newlines.source eq Newlines.keep) && newlines != 0
```

Comment:
> 1. Why `keep` adds newline before lambda argument?

this is the line. if newline is present, the Space split is disabled. (of course, it's up to us to define what `keep` does.)

ah, one more thing which I should have mentioned: the `scala-repos` diffs are not generated by running one version of scalafmt and then another (because the next one can't undo what the first one did). instead, both ran on the same original, pre-scalafmt commit, and the diff is the comparison between results. that's how a space split becomes a newline in `keep`.